PxtlCa.XmlCommentMarkDownGenerator.MSBuild/GenerateMarkdown.cs

```diff
@@ -0,0 +1,273 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+//using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using System.Xml.Linq;
+using System.Xml;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.NamingConventions;
+using PxtlCa.XmlCommentMarkDownGenerator.MSBuild.Options;
+
+namespace PxtlCa.XmlCommentMarkDownGenerator.MSBuild
+{
+    /// <summary>
+    /// A task that generates and optionally merges markdown
+    /// </summary>
+    public class GenerateMarkdown : Task
+    {
+        /// <summary>
+        /// The file(s) from which to generate markdown.  This should be in XmlDocumentation format.
+        /// </summary>
+        [Required]
+        public ITaskItem[] InputXml { get; set; }
+
+        /// <summary>
+        /// DocumentationPath is the top level directory in which to search for files.
+        /// It is also the path where generated markdown files are created.
```

Comment:
This comment highlights the problem of how DocumentationPath wears two hats - it does not make sense that the header docs would be present in the same ./Docs folder they're going to get merged into.